holidays/countries/niger.py

```diff
@@ -0,0 +1,314 @@
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Niger>
+        * <https://web.archive.org/web/20250531032604/https://wageindicator.org/fr-ne/droit-du-travail/les-conges-et-les-jours-de-repos>
+        * <https://web.archive.org/web/20250206194155/https://www.rivermate.com/guides/niger/leave>
+        * [Eid al-Adha](https://web.archive.org/web/20250117013558/https://www.timeanddate.com/holidays/niger/eid-al-adha)
+        * [Eid al-Fitr](https://web.archive.org/web/20250114130118/https://www.timeanddate.com/holidays/niger/eid-al-fitr)
+        * [Laylat al-Qadr](https://web.archive.org/web/20250531033137/https://www.timeanddate.com/holidays/niger/laylat-al-qadr)
+        * [Islamic New Year](https://web.archive.org/web/20240723135601/https://www.timeanddate.com/holidays/niger/muharram-new-year)
+        * [Prophet's Birthday](https://web.archive.org/web/20250124122731/https://www.timeanddate.com/holidays/niger/prophet-birthday)
+        * [Article 114](https://web.archive.org/web/20240205184429/https://droit-afrique.com/upload/doc/niger/Niger-Code-2012-du-travail.pdf)
+
+    Notes:
+        After Law No. 97-020 of June 20, 1997 establishing public holidays came into
+        effect, holidays that fell on the mandatory weekly rest day (Sunday) were
+        observed on the next Monday.
+    """
+
+    country = "NE"
+    default_language = "ha"
+    supported_languages = ("en_US", "ha")
+    # %s (observed).
+    observed_label = tr("%s (lura)")
+    # %s (estimated).
+    estimated_label = tr("%s (kimanta)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (lura, kimanta)")
+    # Niger gain semi-independence from France on Dec 18 1958.
+    start_year = 1959
+    weekend = {SUN}
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=NigerIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        kwargs.setdefault("observed_since", 1998)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("Ranar Sabuwar Shekara")))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Litinin"))
+
+        if self._year >= 1995:
+            # Concord Day.
+            self._add_observed(self._add_holiday_apr_24(tr("Ranar Concord")))
+
+        # International Labor Day.
+        self._add_observed(self._add_labor_day(tr("Ranar Kwadago ta Duniya")))
+
+        # Ascension Thursday.
```

Comment:
According to [this source](https://web.archive.org/web/20250531032604/https://wageindicator.org/fr-ne/droit-du-travail/les-conges-et-les-jours-de-repos),
> The public holidays covered by the interprofessional collective agreement are Ascension, Pentecost, Assumption (August 15) and All Saints' Day (November 1).

As far as I understand, these holidays are not mandatory for everyone. Therefore, they should be classified as OPTIONAL category.